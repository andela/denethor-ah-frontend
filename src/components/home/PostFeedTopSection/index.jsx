import React, { Component } from "react";
import { connect } from 'react-redux';
import PostCard from '../PostCard';
import PropTypes from 'prop-types';
import './style.scss';
import { getArticles } from "../../../redux/actions/articles";

export class PostStreamTopSection extends Component {

  componentDidMount() {
    const { fetchArticles } = this.props;
    fetchArticles();
  }

  render() {
    const sectionArticles = [];
    const { articles, categories } = this.props;

    articles.forEach((item, index) => {
      // stop loading articles after the 4th item
      if(index > 3) return;
  
      const indexToSizeMap = {
        0: 'big',
        1: 'small first',
        2: 'small second',
        3: 'wide',
      };
  
      const size = indexToSizeMap[index];
  
      const postEntry = <PostCard key={item.id} categories={categories} article={item} size={size} />;
  
      sectionArticles.push(postEntry);
    });
  
    return (
      <div className='auto-container'>
        <div className='section-title'>Latest Publications</div>
        <div className='post-stream-top'>
          {sectionArticles}
        </div>
      </div>
    )
  }
}

PostStreamTopSection.propTypes = {
  fetchArticles: PropTypes.func.isRequired,
  articles: PropTypes.array.isRequired,
  categories: PropTypes.array.isRequired
}

const mapStateToProps = (state) => ({
  articles: state.articles,
  categories: state.categories,
});

const mapDispatchToProps = (dispatch) => ({
  fetchArticles: () => dispatch(getArticles())
});

export default connect(mapStateToProps, mapDispatchToProps)(PostStreamTopSection);