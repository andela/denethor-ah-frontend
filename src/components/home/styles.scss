.banner__screen__slide {
  opacity: 0.01;
  transform: scale(0.5) translateX(50%);
}
.banner__screen__slide-active {
  opacity: 1;
  transform: scale(1) translateX(0%);
  transition: all 400ms ease-in-out;
}
.banner__screen__slide-exit {
  opacity: 1;
  transform: scale(1) translateX(0%);
}
.banner__screen__slide-exit-active {
  opacity: 0.01;
  transform: scale(0.5) translateX(50%);
  transition: all 400ms ease-in-out;
}