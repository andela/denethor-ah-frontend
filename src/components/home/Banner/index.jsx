import React from 'react';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';
import Stats from './Stats/Stats';
import Signup from './Signup/Signup';
import Login from './Login/Login';
import './styles.scss';

const sides = {
  Stats, Signup, Login
};

const Banner = ({ side, handleSignup }) => {
  const Side = sides[side];
  return (
    <div className="banner">
      <div className="banner__slogan">
        <div>
          <h1 className="banner__yellow-text hidden-on-mobile">The place <br /> writers love...</h1>
          <h1><span className="banner__yellow-text-mobile">Great writers,</span><br />Quality Content</h1>
          <button onClick={handleSignup}>Get Started</button>
        </div>
      </div>
      <div className="banner__screen">
        <ReactCSSTransitionGroup
          transitionName="background"
          transitionEnterTimeout={500}
          transitionLeaveTimeout={400}
        >
          <Side key={side}/>
        </ReactCSSTransitionGroup>
      </div>
    </div>
  );
};

export default Banner;
