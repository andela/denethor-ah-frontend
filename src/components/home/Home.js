import React from 'react';
import Banner from './Banner';
import './styles.scss';


const Header = (props) => {
  return (
    <div className="home">
      <Banner side={props.side} handleSignup={props.handleSignup} />
    </div>
  )
}

export default Header;