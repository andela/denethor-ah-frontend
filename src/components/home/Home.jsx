import React from 'react';
import Banner from './Banner';


const Home = (props) => {
  return (
    <div className="home">
      <Banner side={props.side} handleSignup={props.handleSignup} />
    </div>
  )
}

export default Home;
