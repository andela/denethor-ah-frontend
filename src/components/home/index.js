import React from 'react';
import Banner from './Banner';
import PostFeedTopSection from './PostFeedTopSection';
import FeedBottom from '../feed';

const Home = (props) => {
  return (
    <div className="home">
      <Banner side={props.side} handleSignup={props.handleSignup} />
      <PostFeedTopSection />
      <FeedBottom />
    </div>
  )
}

export default Home;
