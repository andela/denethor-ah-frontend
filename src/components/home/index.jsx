import React from 'react';
import Banner from './banner';
import { PostFeedTopSection } from '../feed/postFeedTopSection';
import FeedBottom from '../feed';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

const Home = ({ bannerScreen, articles, categories }) => {
  return (
    <div className="home">
      <Banner bannerScreen={bannerScreen} />
      <PostFeedTopSection articles={articles} categories={categories} />
      <FeedBottom />
    </div>
  )
}

Home.propTypes = {
  bannerScreen: PropTypes.node.isRequired,
  articles: PropTypes.array.isRequired,
  categories: PropTypes.array.isRequired,
}

const mapStateToProps = (state) => ({
  articles: state.articles,
  categories: state.categories,
})


export default connect(mapStateToProps)(Home);