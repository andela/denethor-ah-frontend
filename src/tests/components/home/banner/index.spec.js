import React from 'react';
import { shallow } from 'enzyme';
import { Banner } from '../../../../components/home/banner';

test('Should render banner with stats correctly', () => {
  const wrapper = shallow(<Banner bannerScreen={'Stats'} />);
  expect(wrapper).toMatchSnapshot();
});

test('Should call signup function on button click', () => {
  const push = jest.fn();
  const history = { push }
  const wrapper = shallow(<Banner bannerScreen={'Stats'} history={history}/>);
  wrapper.find('button').simulate('click');
  expect(push).toHaveBeenCalledWith('/signup');
});

test('Should render banner with login correctly', () => {
  const wrapper = shallow(<Banner bannerScreen={'Login'}/>);
  expect(wrapper).toMatchSnapshot();
});

test('Should render banner with signup correctly', () => {
  const wrapper = shallow(<Banner bannerScreen={'Signup'}/>);
  expect(wrapper).toMatchSnapshot();
});
