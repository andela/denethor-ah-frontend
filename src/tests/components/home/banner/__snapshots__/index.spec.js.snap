// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render banner with login correctly 1`] = `
<div
  className="banner"
>
  <div
    className="banner__slogan"
  >
    <div>
      <h1
        className="banner__yellow-text hidden-on-mobile"
      >
        The place 
        <br />
         writers love...
      </h1>
      <h1>
        <span
          className="banner__yellow-text-mobile"
        >
          Great writers,
        </span>
        <br />
        Quality Content
      </h1>
      <button
        onClick={[Function]}
      >
        Get Started
      </button>
    </div>
  </div>
  <Screen
    content="Login"
  />
</div>
`;

exports[`Should render banner with signup correctly 1`] = `
<div
  className="banner"
>
  <div
    className="banner__slogan"
  >
    <div>
      <h1
        className="banner__yellow-text hidden-on-mobile"
      >
        The place 
        <br />
         writers love...
      </h1>
      <h1>
        <span
          className="banner__yellow-text-mobile"
        >
          Great writers,
        </span>
        <br />
        Quality Content
      </h1>
      <button
        onClick={[Function]}
      >
        Get Started
      </button>
    </div>
  </div>
  <Screen
    content="Signup"
  />
</div>
`;

exports[`Should render banner with stats correctly 1`] = `
<div
  className="banner"
>
  <div
    className="banner__slogan"
  >
    <div>
      <h1
        className="banner__yellow-text hidden-on-mobile"
      >
        The place 
        <br />
         writers love...
      </h1>
      <h1>
        <span
          className="banner__yellow-text-mobile"
        >
          Great writers,
        </span>
        <br />
        Quality Content
      </h1>
      <button
        onClick={[Function]}
      >
        Get Started
      </button>
    </div>
  </div>
  <Screen
    content="Stats"
  />
</div>
`;
